torch/csrc/jit/jit_log.cpp

```diff
@@ -0,0 +1,63 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <c10/util/Exception.h>
+#include <cstdlib>
+#include <iomanip>
+#include <sstream>
+#include <thread>
+
+namespace torch {
+namespace jit {
+
+JitLoggingLevels jit_log_level() {
+  static const char* c_log_level = std::getenv("PYTORCH_JIT_LOG_LEVEL");
+  static const JitLoggingLevels log_level = c_log_level
+      ? static_cast<JitLoggingLevels>(std::atoi(c_log_level))
+      : JitLoggingLevels::OFF;
+  return log_level;
+}
+
+static std::string level_and_tid(JitLoggingLevels level) {
+  std::stringstream ss;
+  ss << "[" << level << " T";
+  ss << std::this_thread::get_id();
```

Comment:
Is it always useful? It kind of contaminates the output, so I'd make it optional unless it's really useful.

Reply:
that's a great idea. I'll make it optional. Thread ids is very useful when there are multiple concurrent complations which might not be the most common scenario. Log level is also useful when you just want to filter down to the messages at a particular log level.